src/lib/Recorder.ts

```diff
@@ -1,55 +1,64 @@
+/* eslint-disable max-statements */
+/* eslint-disable max-len */
 import PipeSource from './PipeSource';
 
 type TRecorderOptions = {
 	bufferSize?: number;
 	sampleRate?: number;
 	channelCount?: number;
+	publishFrequency?: number;
 };
 
 const defaultOptions: TRecorderOptions = {
 	bufferSize: 1024,
 	sampleRate: 44100,
 	channelCount: 1,
+	publishFrequency: 200,
 };
 
 export default class Recorder extends PipeSource {
 	private readonly options: TRecorderOptions;
 	private isRecording: boolean = false;
 	private micStream: MediaStream;
-	private mediaRecorder: MediaRecorder;
+	private animationID: number;
 
 	constructor(options: TRecorderOptions = {}) {
 		super();
 
 		this.options = { ...defaultOptions, ...options };
 	}
 
+	private listen(analyser: AnalyserNode, data: Float32Array) {
+		analyser.getFloatFrequencyData(data);
+		const positiveData = data.map((frequency) => Math.abs(frequency) / 200);
+		this.publish(positiveData);
+		this.animationID = requestAnimationFrame(() => this.listen(analyser, data));
+	}
+
 	public async start(): Promise<void> {
 		if (!this.micStream) {
 			this.micStream = await navigator.mediaDevices.getUserMedia({
+				video: false,
 				audio: {
 					sampleRate: this.options.sampleRate,
 					channelCount: this.options.channelCount,
 				},
 			});
 		}
-
-		if (!this.mediaRecorder) {
-			this.mediaRecorder = new MediaRecorder(this.micStream);
-
-			this.mediaRecorder.ondataavailable = event => {
-				if (!this.isRecording) return;
-				this.publish(event.data);
-			};
+		if (!this.isRecording) {
+			const context = new AudioContext();
+			const source = context.createMediaStreamSource(this.micStream);
+			const analyser = context.createAnalyser();
+			const data = new Float32Array(analyser.frequencyBinCount);
+			analyser.fftSize = this.options.bufferSize * 2;
+			source.connect(analyser);
+			this.animationID = requestAnimationFrame(() => this.listen(analyser, data));
```

Comment:
this looks redundant. exact same line as above.
just call `this.listen()` here.
if you really need that requestAnimationFrame the very first time it is called, then put that inside the `listen` method and have it wrap a local function instead. which is a bit prettier anyway, I feel.